### PR TITLE
updates on documentation

### PR DESCRIPTION
**Pull request by cdlliuy**

### CODE REVIEW

File: docs/Public_API.rst [hunk 1]
```diff
@@ -1123,12 +1123,22 @@ A JSON payload is required to emit your own metrics with the metric value and th
...
 
       }]
 
   }
 
+<INSTANCE INDEX> is the index of current application instance which would be fetched from environment variable `CF_INSTANCE_INDEX`
+
+<CUSTOM METRIC NAME> is the name of the emit metric which must be equal to the metric name that you define in the policy file. 
+
+<CUSTOM METRIC VALUE> is value that you would like to submit.
```
Comment (qibobo) [inline docs/Public_API.rst]: The value will be marshaled to a float value. So the "CUSTOM METRIC VALUE" must be a number.

File: docs/Public_API.rst [hunk 2]
```diff
@@ -1123,12 +1123,22 @@ A JSON payload is required to emit your own metrics with the metric value and th
...
+
+        "unit": "<CUSTOM METRIC UNIT>"
 
       }]
 
   }
 
+<INSTANCE INDEX> is the index of current application instance which would be fetched from environment variable `CF_INSTANCE_INDEX`
+
+<CUSTOM METRIC NAME> is the name of the emit metric which must be equal to the metric name that you define in the policy file.
```
Comment (qibobo) [inline docs/Public_API.rst]: 1 define->defined?
2 there will be no file if user define policy in ui? So just policy is ok?
3 now in app_metric and appinstancemetrics table the metric_name column is varchar(100), so the length <CUSTOM METRIC NAME> should be <=100;

File: docs/Public_API.rst [hunk 3]
```diff
@@ -1123,12 +1123,22 @@ A JSON payload is required to emit your own metrics with the metric value and th
...
 
   }
 
+<INSTANCE INDEX> is the index of current application instance which would be fetched from environment variable `CF_INSTANCE_INDEX`
+
+<CUSTOM METRIC NAME> is the name of the emit metric which must be equal to the metric name that you define in the policy file. 
+
+<CUSTOM METRIC VALUE> is value that you would like to submit.
+
+<CUSTOM METRIC UNIT> is unit for the metrics, optional.
```
Comment (qibobo) [inline docs/Public_API.rst]: metrics->metric

File: docs/Readme.md [hunk 4]
```diff
@@ -70,16 +69,17 @@ For example, if you want to scale out your application by adding 2 instances whe
...
 
 `cool_down_secs` defines the time duration to wait before the next scaling kicks in.  It helps to ensure that your application does not launch or terminate instances before your application becomes stable. This setting can be configured based on your instance warm-up time or other needs.
 
 *Note:*
 
-You can define multiple scaling-out and scaling-in rules. However, `App-AutoScaler` does not detect conflicts among them.  It is your responsibility to ensure the scaling rules do not conflict with each other to avoid fluctuation or other issues. 
+* You can define multiple scaling-out and scaling-in rules. However, `App-AutoScaler` does not detect conflicts among them.  It is your responsibility to ensure the scaling rules do not conflict with each other to avoid fluctuation or other issues. 
 
+* `breach_duration_secs` and `cool_down_secs` are both optional entries in scaling_rule definition.  The `App Autoscaler` provider will define the default value if you omit them from the policy file.
```
Comment (qibobo) [inline docs/Readme.md]: policy

File: docs/Readme.md [hunk 5]
```diff
@@ -260,19 +260,29 @@ You need to emit  your own metric for scaling to the "URL" specified in credenti
...
+        "unit": "<CUSTOM METRIC UNIT>",
       }
     ]
   }
 ```
 
+*Note:*
+
+* `<INSTANCE INDEX>` is the index of current application instance which would be fetched from environment variable `CF_INSTANCE_INDEX`
+* `<CUSTOM METRIC NAME>` is the name of the emit metric which must be equal to the metric name that you define in the policy file.
```
Comment (qibobo) [inline docs/Readme.md]: The same as above.

File: docs/Public_API.rst (hunk 2 above)
Comment (cdlliuy) [reply docs/Public_API.rst]: 1.`define` should be fine here as the present tense. 
2.  I didn't get your point here ...

File: docs/Readme.md [hunk 6]
```diff
@@ -52,10 +52,9 @@ The following are the built-in metrics that you can use to scale your applicatio
 
 * **custom metric** 
 
-	Custom emtric is supported since [app-autoscaler v3.0.0 release][app-autoscaler-v3.0.0]. You can define your own metric name and emit your own metric to `App Autoscaler` to trigger further dynamic scaling. 
+	Custom emtric is supported since [app-autoscaler v3.0.0 release][app-autoscaler-v3.0.0]. You can define your own metric name and emit your own metric to `App Autoscaler` to trigger further dynamic scaling. Only alphabet letters, numbers and "_" are allowed for a valid metric name, and the maximum length of the metric name is 100 characters.
```
Comment (qibobo) [inline docs/Readme.md]: maximum length of the metric name is 100

**Pull request merged**